highlighters/main/main-highlighter.zsh

```diff
@@ -846,16 +864,9 @@ _zsh_highlight_main_highlighter_highlight_list()
                    fi
                    _zsh_highlight_main__stack_pop 'R' reserved-word
                  fi;;
-        $'\x28\x29') # possibly a function definition
-                 if (( in_redirection )) || $in_array_assignment; then
-                   style=unknown-token
-                 else
-                   if [[ $zsyh_user_options[multifuncdef] == on ]] || false # TODO: or if the previous word was a command word
-                   then
-                     next_word+=':start:'
-                   fi
-                   style=reserved-word
-                 fi
+        '()')
+                 # Function definition was handled above
+                 style=unknown-token
```

Comment:
When can this happen?  `f g ()` with nomultifuncdef, `function "foo'bar" ()`, …?

Reply:
or after a redirection or in an array assignment.

Reply:
What about empty `case` patterns?  `case foo in (bar) ;; esac` → `case foo in () ;; esac`

Reply:
Right now it's highlighted as `reserved-word` which isn't correct either. We'll probably need another state to handle case correctly.

Reply:
With this branch the `()` in `case` is highlighted as unknown-token — which is correct, since it's a syntax error. Let's discuss `case` on #406.